test/e2e/framework/util.go

```diff
@@ -2896,7 +2896,18 @@ func WaitForDeploymentStatusValid(c clientset.Interface, d *extensions.Deploymen
 			Logf(reason)
 			return false, nil
 		}
-		return true, nil
+
+		// When the deployment status and its underlying resources reach the desired state, we're done
+		if deployment.Status.Replicas == deployment.Spec.Replicas &&
```

Comment:
It's somewhere in the perma-failed PR i think:) I will refactor this once I get the API merged (so the impl will get its turn)